community/common/src/test/java/org/neo4j/test/Randoms.java

```diff
@@ -284,14 +298,72 @@ private Object propertyValue( byte type )
         case 7:
             return random.nextDouble();
         case 8:
-            return string();
+            return randomDateTime();
         case 9:
+            return randomTime();
+        case 10:
+            return randomDate();
+        case 11:
+            return randomLocalDateTime();
+        case 12:
+            return randomLocalTime();
+        case 13:
+            return randomPeriod();
+        case 14:
+            return randomDuration();
+        case 15:
+            return string();
+        case 16:
             return array();
         default:
             throw new IllegalArgumentException( "Unknown value type " + type );
         }
     }
 
+    public OffsetTime randomTime()
+    {
+        return OffsetTime.ofInstant( randomInstant(), UTC);
+    }
+
+    public LocalDateTime randomLocalDateTime()
+    {
+        return LocalDateTime.ofInstant( randomInstant(), UTC);
+    }
+
+    public LocalDate randomDate()
+    {
+        return LocalDate.ofEpochDay( nextLong( (-999_999_999L - 1970) * 365, (999_999_999L - 1970) * 365) );
```

Comment:
These calculations require explanations, and possibly breaking out into methods and constants, depending on the explanations. Is this the complete possible span of random dates uniformly distributed? I think that should otherwise be the aim, to be aligned with the general expectations here.